src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -138,7 +119,7 @@ protected Object getObject(BsonValue o, int columnType) throws SQLException {
             case Types.BIGINT:
                 return getInt(o);
             case Types.BINARY:
-                return o.asBinary();
+                return new MongoSQLValue(o.asBinary());
```

Comment:
It does feel a bit strange to only wrap _some_ types in `getObject` with `MongoSQLValue`. It may not be totally clear to future maintainers why some are wrapped (Binary, Decimal, Other, Timestamp) and some are not (all others that do not throw an exception). The distinction right now is that anything that needs special extended json stringification is wrapped. Not sure if that's the best criteria. Maybe we should wrap everything?